File: src/typescript/src/overlay.test.ts
import { Overlay } from './overlay';

describe('Overlay', () => {
  describe('OverlayConfig defaults', () => {
    it('should have correct default values', () => {
      const overlay = new Overlay();
      const config = overlay.getConfig();
      expect(config.enabled).toBe(true);
      expect(config.opacity).toBe(0.9);
      expect(config.width).toBe(800);
      expect(config.height).toBe(600);
    });
  });

  describe('visibility', () => {
    it('should start hidden', () => {
      const overlay = new Overlay();
      expect(overlay.isVisible()).toBe(false);
    });

    it('should show overlay when enabled', () => {
      const overlay = new Overlay();
      const result = overlay.show();
      expect(result).toBe(true);
      expect(overlay.isVisible()).toBe(true);
    });

    it('should not show overlay when disabled', () => {
      const overlay = new Overlay({ enabled: false });
      const result = overlay.show();
      expect(result).toBe(false);
      expect(overlay.isVisible()).toBe(false);
    });

    it('should hide overlay', () => {
      const overlay = new Overlay();
      overlay.show();
      const result = overlay.hide();
      expect(result).toBe(true);
      expect(overlay.isVisible()).toBe(false);
    });

    it('should toggle visibility', () => {
      const overlay = new Overlay();
      overlay.toggle();
      expect(overlay.isVisible()).toBe(true);
      overlay.toggle();
      expect(overlay.isVisible()).toBe(false);
    });
  });
});

File: src/go/internal/overlay/overlay_test.go
package overlay

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestDefaultConfig(t *testing.T) {
	config := DefaultConfig()
	assert.True(t, config.Enabled)
	assert.InDelta(t, 0.9, config.Opacity, 0.001)
	assert.Equal(t, 800, config.Width)
	assert.Equal(t, 600, config.Height)
}

func TestNewOverlay(t *testing.T) {
	o := New(nil)
	assert.NotNil(t, o)
	assert.False(t, o.IsVisible())
}

func TestShowOverlay(t *testing.T) {
	o := New(nil)
	result := o.Show()
	assert.True(t, result)
	assert.True(t, o.IsVisible())
}

func TestShowDisabledOverlay(t *testing.T) {
	config := &Config{Enabled: false}
	o := New(config)
	result := o.Show()
	assert.False(t, result)
	assert.False(t, o.IsVisible())
}

func TestHideOverlay(t *testing.T) {
	o := New(nil)
	o.Show()
	result := o.Hide()
	assert.True(t, result)
	assert.False(t, o.IsVisible())
}

func TestToggleOverlay(t *testing.T) {
	o := New(nil)
	o.Toggle()
	assert.True(t, o.IsVisible())
	o.Toggle()
	assert.False(t, o.IsVisible())
}
